pyecoregen/ecore.py

```diff
@@ -237,6 +237,8 @@ def create_environment(self, **kwargs):
 
         return environment
 
-    def generate(self, model, outfolder):
+    def generate(self, model, outfolder, global_vars=None):
+        if global_vars:
+            self.environment.globals.update(global_vars)
```

Comment:
I don't quite like this polluting the template context a bit (with all the command line arguments). Could you make this an explicitly named argument instead?

Then please extend the environment in `create_environment`. This probably requires a generator instance variable...

At the same time, I am not too happy about using `environment` for this. What you in fact do is pass a template-specific variable to the package template. Something like this should go into the template's context, and only for the template that needs it.

This is not possible with the current design (from my side). There should be a way to pass in context data from the generator as well. I have to think about this.

So let's do this:

* You just rename the argument, but leave the code where it is (in `generate`).
* I will introduce a way to pass context data from generator to tasks and then move your parameter there.

Reply:
That's good for me, I wasn't sure about using the `environement`, thanks for the code review! There is no emergency about this, I thought this little addition could be interesting, but clearly it will probably serves few people.

Regarding the instance variable, it sounds a little bit strange to me as the generator takes two arguments: the model and the output folder, and the fact that in a programmatic context, the generator could be used many times on many input models with different output folders. Consequently, a single generation produced by the generator depends on a duet `(model, output-folder)`. If we add the 'auto-registration' in the equation, a single generation would depend on the triplet: `(model, output-folder, auto-registration)` as for two different generation with the same generator instance, the parameter can be either `True` or `False` regarding what the user want and the input model.